frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/model/ResourceDescriptor.java

```diff
@@ -143,35 +163,113 @@ public static ResourceDescriptor fromResource(Resource res) throws Exception {
     } catch (IllegalAccessException e) {
       LOGGER.logError(e, "Ignore: Fail to get GPU information, illegal access function");
     }
+
+    try {
+
+      Class<?> clazz = res.getClass();
+      Class hadoopValueRangesClass = Class.forName("org.apache.hadoop.yarn.api.records.ValueRanges");
+      Class hadoopValueRangeClass = Class.forName("org.apache.hadoop.yarn.api.records.ValueRange");
```

Comment:
>ValueRange [](start = 86, length = 10)

We would better rename the class Range to ValueRange. Align with hadoop interface